gdnative-core/src/geom/basis.rs

```diff
@@ -19,8 +25,46 @@ impl Basis {
         unsafe { std::mem::transmute::<sys::godot_basis, Self>(c) }
     }
 
-    pub fn identity() -> Basis {
+    /// The basis that will flip something along the X axis when used in a transformation.
+    pub const fn flip_x() -> Basis {
+        Basis::from_diagonal(Vector3::new(-1.0, 1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the Y axis when used in a transformation.
+    pub const fn flip_y() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, -1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the Z axis when used in a transformation.
+    pub const fn flip_z() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, 1.0, -1.0))
+    }
+
+    pub const fn from_diagonal(p_diag: Vector3) -> Basis {
         Basis {
+            elements: [
+                Vector3::new(p_diag.x, 0.0, 0.0),
+                Vector3::new(0.0, p_diag.y, 0.0),
+                Vector3::new(0.0, 0.0, p_diag.z),
+            ],
+        }
+    }
+
+    /// Creates a Basis from 3 `Vectors3`s
+    pub fn from_elements(elements: [Vector3; 3]) -> Self {
+        Self { elements }
+    }
+
+    /// Create a rotation matrix (in the YXZ convention: first Z, then X, and Y last) from the specified Euler angles, given in the vector format as (X angle, Y angle, Z angle).
+    pub fn from_euler(euler: Vector3) -> Self {
+        let mut b = Basis::default();
+        b.set_euler_yxz(&euler);
+        b
+    }
+
+    /// The identity basis. This is identical to calling Basis::default(). This constant can be used to make your code clearer.
```

Comment:
Code and identifiers in doc-comments should be surrounded by `` ` ``, since they're parsed as Markdown. e.g.: `` `Basis::default()` ``.